libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -410,6 +410,47 @@ bool ExpressionCompiler::visit(TupleExpression const& _tuple)
 bool ExpressionCompiler::visit(UnaryOperation const& _unaryOperation)
 {
 	CompilerContext::LocationSetter locationSetter(m_context, _unaryOperation);
+
+	if (_unaryOperation.annotation().userDefinedFunction)
+	{
+		FunctionDefinition const& function = *_unaryOperation.annotation().userDefinedFunction;
+		FunctionType const* functionType = dynamic_cast<FunctionType const*>(
+			function.libraryFunction() ? function.typeViaContractName() : function.type());
+
+		solAssert(functionType);
+		functionType = dynamic_cast<FunctionType const&>(*functionType).asBoundFunction();
+		solAssert(functionType);
+		evmasm::AssemblyItem returnLabel = m_context.pushNewTag();
+		_unaryOperation.subExpression().accept(*this);
+		utils().pushCombinedFunctionEntryLabel(
```

Comment:
Since we are calling the function right away, I think we can directly push the correct function label instead of using the combined one.

Reply:
Code adjusted as suggested.